extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -71,234 +147,135 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 			connectionInfo.database = selectedDatabase;
 
 			// prompt user for object name to create function from
+			telemetryStep = 'getObjectName';
 			objectName = await azureFunctionsUtils.promptForObjectName(selectedBinding.type);
 			if (!objectName) {
 				// user cancelled
 				return;
 			}
-
-		} catch (e) {
-			propertyBag.quickPickStep = quickPickStep;
-			exitReason = 'error';
-			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick, undefined, utils.getErrorType(e))
-				.withAdditionalProperties(propertyBag).send();
-			return;
-		}
-	} else {
-		quickPickStep = 'launchFromTable';
-		connectionInfo = node.connectionInfo;
-		// set the database containing the selected table so it can be used
-		// for the initial catalog property of the connection string
-		let newNode: ITreeNodeInfo = node;
-		while (newNode) {
-			if (newNode.nodeType === 'Database') {
-				connectionInfo.database = newNode.metadata.name;
-				break;
-			} else {
-				newNode = newNode.parentNode;
+		} else {
+			// if user selects table in tree view we use connection info from Object Explorer node
+			telemetryStep = 'launchFromTable';
+			connectionInfo = node.connectionInfo;
+			// set the database containing the selected table so it can be used
+			// for the initial catalog property of the connection string
+			let newNode: ITreeNodeInfo = node;
+			while (newNode) {
```

Comment:
This logic to get the database name should go in a separate helper function - it's generic enough (and this function is massive enough as it is)